packages/wonder-blocks-modal/src/components/__tests__/flexible-dialog.test.tsx

```diff
@@ -0,0 +1,253 @@
+                    name: "Mystery mission",
+                }),
+            ).toBeInTheDocument();
+        });
+
+        it("applies title component in a content slot as the accessible name for a modal", () => {
+            // Arrange
+            render(
+                <FlexibleDialog
+                    title={<Heading>Scooby Doo</Heading>}
+                    content={({title}) => (
+                        <div>
+                            {title}
+                            <p id="description">cool dialog</p>
+                        </div>
+                    )}
+                />,
+            );
+
+            // Act
+            const dialog = screen.getByRole("dialog");
+
+            // Assert
+            expect(dialog).toHaveAccessibleName("Scooby Doo");
+        });
+
+        it("applies aria-label to the modal", () => {
+            // Arrange
+            render(
+                <FlexibleDialog
+                    aria-label="Accessible name"
+                    content={<p id="description">cool dialog</p>}
+                />,
+            );
+
+            // Act
+            const dialog = screen.getByRole("dialog");
+
+            // Assert
+            expect(dialog).toHaveAccessibleName("Accessible name");
+        });
+
+        it("applies aria-labelledby from titleId and title component", () => {
+            // Arrange
+            render(
+                <FlexibleDialog
+                    title={<Heading>Mystery missionz</Heading>}
+                    titleId="heading-1"
+                    content={<p id="description">cool dialog</p>}
+                />,
+            );
+
+            // Act
+            const dialog = screen.getByRole("dialog");
+
+            // Assert
+            expect(dialog).toHaveAccessibleName("Mystery missionz");
+        });
+
+        it("binds aria-labelledby to a title with its own id", () => {
+            // Arrange
+            render(
+                <FlexibleDialog
+                    title={<Heading id="a-heading">Mystery mission 2</Heading>}
+                    aria-labelledby="a-heading"
+                    content={<p id="description">cool dialog</p>}
+                />,
+            );
+
+            // Act
+            const dialog = screen.getByRole("dialog");
+
+            // Assert
+            expect(dialog).toHaveAccessibleName("Mystery mission 2");
+        });
+
+        it("applies aria-describedby to the modal", () => {
+            // Arrange
+            render(
+                <FlexibleDialog
+                    title="unused"
+                    content={<BodyText id="description">cool dialog</BodyText>}
+                    aria-describedby="description"
+                />,
+            );
+
+            // Act
+            const modal = screen.getByRole("dialog");
+
+            // Assert
+            expect(modal).toHaveAccessibleDescription(/cool dialog/i);
+        });
+    });
+});
```

Comment:
Would it be helpful to also include jest-axe tests? We have `toHaveNoA11yViolations` matcher for running axe checks on components!

Reply:
I added this! I'm hesitant to add a bunch of these for all the label combinations for perf reasons (axe is a bit heavy for unit tests). Hopefully the types cover it, though.